### PR TITLE
added maria-db

### PR DESCRIPTION
**Pull request by raffaelespazzoli**
#### What is this PR About?
A demo of mariadb installed in HA multi-master using galera.
Currently using petsets, will need to be upgraded to stateful sets in 3.5
Potential improvements: 
 - improve data source parameter to increase availability
 - the chaos monkey pod stopped working. version should probably be  downgraded to a working one. 

#### How do we test this?
Follow the readme

cc: @redhat-cop/containerize-it

### CODE REVIEW

File: mariadb-ha/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,17 @@
+FROM registry.access.redhat.com/rhscl/mariadb-101-rhel7
+
+USER root
+
+RUN yum clean all && \
+    yum-config-manager --enable rhel-server-rhscl-7-rpms && \
+    yum-config-manager --enable rhel-7-server-optional-rpms && \
+    INSTALL_PKGS="rh-mariadb101-galera which" && \
```
Comment (etsauer) [inline mariadb-ha/Dockerfile]: When my build runs I get the following:

```
No package rh-mariadb101-galera available.
```

Where should the `rh-mariadb101-galera` package come form?

Comment (raffaelespazzoli) [inline mariadb-ha/Dockerfile]: the build works fine for me. I use the following pool when I build my cluster `8a85f9843e3d687a013e3ddd46dd07f1`

here is the log of my build:
```
Cloning "https://github.com/raffaelespazzoli/containers-quickstarts" ...
	Commit:	ad256f08781712447738efbeff2a2f703ef21691 (Update README.md)
	Author:	raffaelespazzoli <raffaele.spazzoli@gmail.com>
	Date:	Thu Nov 24 09:42:59 2016 -0500

Pulling image registry.access.redhat.com/rhscl/mariadb-101-rhel7@sha256:86ee10d39753b2d54503b6570a4511fbe5581b87be7db77b8f9517abbab355fb ...
Pulled 2/3 layers, 67% complete
Pulled 3/3 layers, 100% complete
Extracting
Step 1 : FROM registry.access.redhat.com/rhscl/mariadb-101-rhel7@sha256:86ee10d39753b2d54503b6570a4511fbe5581b87be7db77b8f9517abbab355fb
 ---> 323f01b29681
Step 2 : USER root
 ---> Running in 87d7175081e9
 ---> 3041db5d8e2a
Removing intermediate container 87d7175081e9
Step 3 : RUN yum clean all &&     yum-config-manager --enable rhel-server-rhscl-7-rpms &&     yum-config-manager --enable rhel-7-server-optional-rpms &&     INSTALL_PKGS="rh-mariadb101-galera which" &&     yum install -y --setopt=tsflags=nodocs $INSTALL_PKGS &&     rpm -V $INSTALL_PKGS &&     yum clean all
 ---> Running in 697b7b5a898d
Loaded plugins: ovl, product-id, search-disabled-repos, subscription-manager
Cleaning repos: rhel-7-fast-datapath-htb-rpms rhel-7-fast-datapath-rpms
              : rhel-7-server-aus-rpms rhel-7-server-eus-rpms
              : rhel-7-server-htb-rpms rhel-7-server-nfv-rpms rhel-7-server-rpms
              : rhel-7-server-rt-beta-rpms rhel-7-server-rt-htb-rpms
              : rhel-7-server-rt-rpms rhel-7-server-tus-rpms
              : rhel-ha-for-rhel-7-server-eus-rpms
              : rhel-ha-for-rhel-7-server-htb-rpms
              : rhel-ha-for-rhel-7-server-rpms
              : rhel-lb-for-rhel-7-server-htb-rpms
              : rhel-rs-for-rhel-7-server-eus-rpms
              : rhel-rs-for-rhel-7-server-htb-rpms
              : rhel-rs-for-rhel-7-server-rpms
              : rhel-sap-for-rhel-7-server-eus-rpms
              : rhel-sap-for-rhel-7-server-rpms
              : rhel-sap-hana-for-rhel-7-server-eus-rpms
              : rhel-sap-hana-for-rhel-7-server-rpms
              : rhel-sjis-for-rhel-7-server-aus-debug-rpms
              : rhel-sjis-for-rhel-7-server-aus-rpms
              : rhel-sjis-for-rhel-7-server-aus-source-rpms
              : rhel-sjis-for-rhel-7-server-debug-rpms
              : rhel-sjis-for-rhel-7-server-rpms
              : rhel-sjis-for-rhel-7-server-source-rpms
Cleaning up everything
Loaded plugins: ovl, product-id
======================== repo: rhel-server-rhscl-7-rpms ========================
[rhel-server-rhscl-7-rpms]
async = True
bandwidth = 0
base_persistdir = /var/lib/yum/repos/x86_64/7Server
baseurl = https://cdn.redhat.com/content/dist/rhel/server/7/7Server/x86_64/rhscl/1/os
cache = 0
cachedir = /var/cache/yum/x86_64/7Server/rhel-server-rhscl-7-rpms
check_config_file_age = True
compare_providers_priority = 80
cost = 1000
deltarpm_metadata_percentage = 100
deltarpm_percentage = 
enabled = 1
enablegroups = True
exclude = 
failovermethod = priority
ftp_disable_epsv = False
gpgcadir = /var/lib/yum/repos/x86_64/7Server/rhel-server-rhscl-7-rpms/gpgcadir
gpgcakey = 
gpgcheck = True
gpgdir = /var/lib/yum/repos/x86_64/7Server/rhel-server-rhscl-7-rpms/gpgdir
gpgkey = file:///etc/pki/rpm-gpg/RPM-GPG-KEY-redhat-release,
   file:///etc/pki/rpm-gpg/RPM-GPG-KEY-redhat-release
hdrdir = /var/cache/yum/x86_64/7Server/rhel-server-rhscl-7-rpms/headers
http_caching = all
includepkgs = 
ip_resolve = 
keepalive = True
keepcache = False
mddownloadpolicy = sqlite
mdpolicy = group:small
mediaid = 
metadata_expire = 86400
metadata_expire_filter = read-only:present
metalink = 
minrate = 0
mirrorlist = 
mirrorlist_expire = 86400
name = Red Hat Software Collections RPMs for Red Hat Enterprise Linux 7 Server
old_base_cache_dir = 
password = 
persistdir = /var/lib/yum/repos/x86_64/7Server/rhel-server-rhscl-7-rpms
pkgdir = /var/cache/yum/x86_64/7Server/rhel-server-rhscl-7-rpms/packages
proxy = False
proxy_dict = 
proxy_password = 
proxy_username = 
repo_gpgcheck = False
retries = 10
skip_if_unavailable = False
ssl_check_cert_permissions = True
sslcacert = /etc/rhsm-host/ca/redhat-uep.pem
sslclientcert = /etc/pki/entitlement-host/8389446063796068379.pem
sslclientkey = /etc/pki/entitlement-host/8389446063796068379-key.pem
sslverify = True
throttle = 0
timeout = 30.0
ui_id = rhel-server-rhscl-7-rpms/7Server/x86_64
ui_repoid_vars = releasever,
   basearch
username = 

Loaded plugins: ovl, product-id
====================== repo: rhel-7-server-optional-rpms =======================
[rhel-7-server-optional-rpms]
async = True
bandwidth = 0
base_persistdir = /var/lib/yum/repos/x86_64/7Server
baseurl = https://cdn.redhat.com/content/dist/rhel/server/7/7Server/x86_64/optional/os
cache = 0
cachedir = /var/cache/yum/x86_64/7Server/rhel-7-server-optional-rpms
check_config_file_age = True
compare_providers_priority = 80
cost = 1000
deltarpm_metadata_percentage = 100
deltarpm_percentage = 
enabled = 1
enablegroups = True
exclude = 
failovermethod = priority
ftp_disable_epsv = False
gpgcadir = /var/lib/yum/repos/x86_64/7Server/rhel-7-server-optional-rpms/gpgcadir
gpgcakey = 
gpgcheck = True
gpgdir = /var/lib/yum/repos/x86_64/7Server/rhel-7-server-optional-rpms/gpgdir
gpgkey = file:///etc/pki/rpm-gpg/RPM-GPG-KEY-redhat-release
hdrdir = /var/cache/yum/x86_64/7Server/rhel-7-server-optional-rpms/headers
http_caching = all
includepkgs = 
ip_resolve = 
keepalive = True
keepcache = False
mddownloadpolicy = sqlite
mdpolicy = group:small
mediaid = 
metadata_expire = 86400
metadata_expire_filter = read-only:present
metalink = 
minrate = 0
mirrorlist = 
mirrorlist_expire = 86400
name = Red Hat Enterprise Linux 7 Server - Optional (RPMs)
old_base_cache_dir = 
password = 
persistdir = /var/lib/yum/repos/x86_64/7Server/rhel-7-server-optional-rpms
pkgdir = /var/cache/yum/x86_64/7Server/rhel-7-server-optional-rpms/packages
proxy = False
proxy_dict = 
proxy_password = 
proxy_username = 
repo_gpgcheck = False
retries = 10
skip_if_unavailable = False
ssl_check_cert_permissions = True
sslcacert = /etc/rhsm-host/ca/redhat-uep.pem
sslclientcert = /etc/pki/entitlement-host/8389446063796068379.pem
sslclientkey = /etc/pki/entitlement-host/8389446063796068379-key.pem
sslverify = True
throttle = 0
timeout = 30.0
ui_id = rhel-7-server-optional-rpms/7Server/x86_64
ui_repoid_vars = releasever,
   basearch
username = 

Loaded plugins: ovl, product-id, search-disabled-repos, subscription-manager
Resolving Dependencies
--> Running transaction check
---> Package rh-mariadb101-galera.x86_64 0:25.3.12-9.el7 will be installed
--> Processing Dependency: nmap-ncat for package: rh-mariadb101-galera-25.3.12-9.el7.x86_64
--> Processing Dependency: libboost_program_options.so.1.53.0()(64bit) for package: rh-mariadb101-galera-25.3.12-9.el7.x86_64
---> Package which.x86_64 0:2.20-7.el7 will be installed
--> Running transaction check
---> Package boost-program-options.x86_64 0:1.53.0-26.el7 will be installed
---> Package nmap-ncat.x86_64 2:6.40-7.el7 will be installed
--> Processing Dependency: libpcap.so.1()(64bit) for package: 2:nmap-ncat-6.40-7.el7.x86_64
--> Running transaction check
---> Package libpcap.x86_64 14:1.5.3-8.el7 will be installed
--> Finished Dependency Resolution

Dependencies Resolved

================================================================================
 Package                Arch    Version         Repository                 Size
================================================================================
Installing:
 rh-mariadb101-galera   x86_64  25.3.12-9.el7   rhel-server-rhscl-7-rpms  1.3 M
 which                  x86_64  2.20-7.el7      rhel-7-server-aus-rpms     41 k
Installing for dependencies:
 boost-program-options  x86_64  1.53.0-26.el7   rhel-7-server-htb-rpms    156 k
 libpcap                x86_64  14:1.5.3-8.el7  rhel-7-server-aus-rpms    138 k
 nmap-ncat              x86_64  2:6.40-7.el7    rhel-7-server-aus-rpms    201 k

Transaction Summary
================================================================================
Install  2 Packages (+3 Dependent packages)

Total download size: 1.8 M
Installed size: 5.4 M
Downloading packages:
warning: /var/cache/yum/x86_64/7Server/rhel-server-rhscl-7-rpms/packages/rh-mariadb101-galera-25.3.12-9.el7.x86_64.rpm: Header V3 RSA/SHA256 Signature, key ID fd431d51: NOKEY
Public key for rh-mariadb101-galera-25.3.12-9.el7.x86_64.rpm is not installed
Public key for boost-program-options-1.53.0-26.el7.x86_64.rpm is not installed
Public key for libpcap-1.5.3-8.el7.x86_64.rpm is not installed
--------------------------------------------------------------------------------
Total                                              1.2 MB/s | 1.8 MB  00:01     
Retrieving key from file:///etc/pki/rpm-gpg/RPM-GPG-KEY-redhat-release
Importing GPG key 0xFD431D51:
 Userid     : "Red Hat, Inc. (release key 2) <security@redhat.com>"
 Fingerprint: 567e 347a d004 4ade 55ba 8a5f 199e 2f91 fd43 1d51
 Package    : redhat-release-server-7.3-7.el7.x86_64 (@anaconda/7.3)
 From       : /etc/pki/rpm-gpg/RPM-GPG-KEY-redhat-release
Importing GPG key 0x2FA658E0:
 Userid     : "Red Hat, Inc. (auxiliary key) <security@redhat.com>"
 Fingerprint: 43a6 e49c 4a38 f4be 9abf 2a53 4568 9c88 2fa6 58e0
 Package    : redhat-release-server-7.3-7.el7.x86_64 (@anaconda/7.3)
 From       : /etc/pki/rpm-gpg/RPM-GPG-KEY-redhat-release
Running transaction check
Running transaction test
Transaction test succeeded
Running transaction
  Installing : boost-program-options-1.53.0-26.el7.x86_64                   1/5 
  Installing : 14:libpcap-1.5.3-8.el7.x86_64                                2/5 
  Installing : 2:nmap-ncat-6.40-7.el7.x86_64                                3/5 
  Installing : rh-mariadb101-galera-25.3.12-9.el7.x86_64                    4/5 
  Installing : which-2.20-7.el7.x86_64                                      5/5 
install-info: No such file or directory for /usr/share/info/which.info.gz
  Verifying  : 14:libpcap-1.5.3-8.el7.x86_64                                1/5 
  Verifying  : boost-program-options-1.53.0-26.el7.x86_64                   2/5 
  Verifying  : which-2.20-7.el7.x86_64                                      3/5 
  Verifying  : 2:nmap-ncat-6.40-7.el7.x86_64                                4/5 
  Verifying  : rh-mariadb101-galera-25.3.12-9.el7.x86_64                    5/5 

Installed:
  rh-mariadb101-galera.x86_64 0:25.3.12-9.el7     which.x86_64 0:2.20-7.el7    

Dependency Installed:
  boost-program-options.x86_64 0:1.53.0-26.el7   libpcap.x86_64 14:1.5.3-8.el7  
  nmap-ncat.x86_64 2:6.40-7.el7                 

Complete!
Loaded plugins: ovl, product-id, search-disabled-repos, subscription-manager
Cleaning repos: rhel-7-fast-datapath-htb-rpms rhel-7-fast-datapath-rpms
              : rhel-7-server-aus-rpms rhel-7-server-eus-rpms
              : rhel-7-server-htb-rpms rhel-7-server-nfv-rpms
              : rhel-7-server-optional-rpms rhel-7-server-rpms
              : rhel-7-server-rt-beta-rpms rhel-7-server-rt-htb-rpms
              : rhel-7-server-rt-rpms rhel-7-server-tus-rpms
              : rhel-ha-for-rhel-7-server-eus-rpms
              : rhel-ha-for-rhel-7-server-htb-rpms
              : rhel-ha-for-rhel-7-server-rpms
              : rhel-lb-for-rhel-7-server-htb-rpms
              : rhel-rs-for-rhel-7-server-eus-rpms
              : rhel-rs-for-rhel-7-server-htb-rpms
              : rhel-rs-for-rhel-7-server-rpms
              : rhel-sap-for-rhel-7-server-eus-rpms
              : rhel-sap-for-rhel-7-server-rpms
              : rhel-sap-hana-for-rhel-7-server-eus-rpms
              : rhel-sap-hana-for-rhel-7-server-rpms rhel-server-rhscl-7-rpms
              : rhel-sjis-for-rhel-7-server-aus-debug-rpms
              : rhel-sjis-for-rhel-7-server-aus-rpms
              : rhel-sjis-for-rhel-7-server-aus-source-rpms
              : rhel-sjis-for-rhel-7-server-debug-rpms
              : rhel-sjis-for-rhel-7-server-rpms
              : rhel-sjis-for-rhel-7-server-source-rpms
Cleaning up everything
 ---> 5f0ab0333fcb
Removing intermediate container 697b7b5a898d
Step 4 : USER 27
 ---> Running in 2aa0ad6bf6d1
 ---> d5765ac27be9
Removing intermediate container 2aa0ad6bf6d1
Step 5 : ADD root /
 ---> c72b194c417e
Removing intermediate container 9a29589f935e
Step 6 : EXPOSE 4567 4568 4444
 ---> Running in 0a0d9e604064
 ---> b0db55c0f8e1
Removing intermediate container 0a0d9e604064
Step 7 : ENV "OPENSHIFT_BUILD_NAME" "mariadb-ha-1" "OPENSHIFT_BUILD_NAMESPACE" "mariadb-ha" "OPENSHIFT_BUILD_SOURCE" "https://github.com/raffaelespazzoli/containers-quickstarts" "OPENSHIFT_BUILD_REFERENCE" "mariadb-ha" "OPENSHIFT_BUILD_COMMIT" "ad256f08781712447738efbeff2a2f703ef21691"
 ---> Running in ff00f6ae9459
 ---> ca4f2652cd2f
Removing intermediate container ff00f6ae9459
Step 8 : LABEL "io.openshift.build.commit.message" "Update README.md" "io.openshift.build.source-location" "https://github.com/raffaelespazzoli/containers-quickstarts" "io.openshift.build.source-context-dir" "mariadb-ha" "io.openshift.build.commit.author" "raffaelespazzoli \u003craffaele.spazzoli@gmail.com\u003e" "io.openshift.build.commit.date" "Thu Nov 24 09:42:59 2016 -0500" "io.openshift.build.commit.id" "ad256f08781712447738efbeff2a2f703ef21691" "io.openshift.build.commit.ref" "mariadb-ha"
 ---> Running in b3d6aefd48c7
 ---> 9cda4c19fcaa
Removing intermediate container b3d6aefd48c7
Successfully built 9cda4c19fcaa

Pushing image 172.30.224.83:5000/mariadb-ha/mariadb-ha:latest ...
Pushed 2/5 layers, 60% complete
Pushed 3/5 layers, 71% complete
Pushed 4/5 layers, 85% complete
Pushed 5/5 layers, 100% complete
Push successful
```

File: mariadb-ha/README.md [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+
+The client can connect to either instances or all of them if it is managing a connection pool.
+
+This architecture can be implemented in OpenShift with PetSet.
+
+The following diagram represent this approach.
+
+![](media/Mariadb-ha-petset.png "")
+
+Note: This project uses PetSets currently not in tech preview (i.e. not supported) in OpenShift.
```
Comment (raffaelespazzoli) [inline mariadb-ha/README.md]: this is fixed.

**Comment by sabre1041** [comment]
@raffaelespazzoli This is quite old and before the deprecation of PetSets which results in the failure upon object creation.

There area also several improvements that would need to be made to the _README_  to reference the upstream _containers-quickstarts_ repository instead of your repo/branch

**Comment by etsauer** [comment]
closing as it is very old.

**Pull request closed without merging**